### PR TITLE
DevEnv: updates nodejs from 10.x to 12.x and golang to 1.13 in ci-deploy dockerfile.

### LINKED ISSUE #19668: DevEnv: Update package.json and Dockerfile to use Node 12.x.
**What would you like to be added**:

Update package.json and Dockerfile to use Node 12.x as it becomes an Active LTS on 2019-10-22.

**Why is this needed**:
To keep Nodejs on a current Active LTS version.

### LINKED ISSUE #20302: Update allowed Node version of node to 12.13.x
**What would you like to be added**:
We don't allow node after 11.0.0 via `package.json`, however newest LTS is 12.13.0.

**Why is this needed**:
To follow Node release cycles.

### LINKED ISSUE #20418: Build: Upgrade Go to 1.13.4
Grafana currently builds against go 1.13.1. The golang team have released 1.13.4: https://groups.google.com/forum/#!topic/golang-announce/YVXawNxmEBw

- go1.13.2 (released 2019/10/17) includes security fixes to the crypto/dsa package and the compiler. See the Go 1.13.2 milestone on our issue tracker for details.
- go1.13.3 (released 2019/10/17) includes fixes to the go command, the toolchain, the runtime, syscall, net, net/http, and crypto/ecdsa packages. See the Go 1.13.3 milestone on our issue tracker for details.
- go1.13.4 (released 2019/10/31) includes fixes to the net/http and syscall packages. It also fixes an issue on macOS 10.15 Catalina where the non-notarized installer and binaries were being rejected by Gatekeeper. See the Go 1.13.4 milestone on our issue tracker for details.

Reference: https://golang.org/doc/devel/release.html#go1.13.minor

### PR DESCRIPTION
**Pull request by Estrax**
**What this PR does / why we need it**:

Updates nodejs from 10.x to the new LTS, version 12.x. Also, updates golang from 1.11 to 1.13 in ci-deploy dockerfile.

**Which issue(s) this PR fixes**:

Fixes #19668 
Fixes #20302 
Closes #20418

### CODE REVIEW

**Comment by Estrax** [comment]
cc @marefr @hugohaggmark @dprokop

**Comment by hugohaggmark** [comment]
@Estrax great work, did the pre-commit hook work? And was it any help?

**Comment by Estrax** [comment]
@hugohaggmark yep, worked flawlessly after reinstalling dependencies under node 12.x.
Screenshot

**Comment by hugohaggmark** [comment]
> @hugohaggmark yep, worked flawlessly after reinstalling dependencies under node 12.x.
> Screenshot

Awesome, did you change in https://github.com/grafana/grafana-build-container as well?

**Comment by Estrax** [comment]
@hugohaggmark no, because based on its readme it has been moved to the core Grafana repo; so I updated https://github.com/grafana/grafana/tree/master/scripts/build/ci-build instead.

**Comment by hugohaggmark** [comment]
> @hugohaggmark no, because based on its readme it has been moved to the core Grafana repo; so I updated https://github.com/grafana/grafana/tree/master/scripts/build/ci-build instead.

Ok cool, can you update the comment in the pre-commit hook then to be up to date?

**Comment by dprokop** [comment]
@Estrax also fixes #20302

**Comment by dprokop** [comment]
@Estrax seems like the build is failing, I think build-container image needs to be updated as well as @hugohaggmark mentioned

**Review by aknuds1 — APPROVED** [review]
LGTM, once the build is fixed.

**Comment by Estrax** [comment]
@hugohaggmark Grafana build container updated.

File: scripts/build/ci-build/Dockerfile [hunk 1]
```diff
@@ -71,10 +71,10 @@ RUN apt-get update   && \
 # base image to crossbuild grafana
 FROM ubuntu:14.04
 
-ENV GOVERSION=1.13.1 \
+ENV GOVERSION=1.13.4 \
```
Comment (marefr) [inline scripts/build/ci-build/Dockerfile]: Due to changes in this file you'll need to update/increment buildcontainer version and push to docker hub, see https://github.com/grafana/grafana/blob/master/scripts/build/ci-build/build-deploy.sh circleci. When that's done you want to update all references to grafana/build-container in the repository.

File: scripts/build/ci-deploy/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM circleci/golang:1.11
+FROM circleci/golang:1.13
```
Comment (marefr) [inline scripts/build/ci-deploy/Dockerfile]: Due to changes in this file you'll need to update/increment  version and push to docker hub, see .https://github.com/grafana/grafana/blob/master/scripts/build/ci-deploy/build-deploy.sh When that's done you want to update all references to grafana/grafana-ci-deploy in the repository.

**Review by marefr — CHANGES_REQUESTED** [review]
Looks good, but need to build and push docker images to docker hub so that we can utilize new versions when building/releasing Grafana in circleci. See comments for details. 

I think after this one is done there's also some changes needed in the enterprise repo. 

Let us know if you need help with building/pushing to docker hub or anything else.

**Comment by marefr** [comment]
Missed the earlier comments about build container and failing builds. I'll create an issue to track go upgrade to 1.13.4

**Comment by Estrax** [comment]
@marefr updated references to `grafana/grafana-ci-build` and `grafana/grafana-ci-deploy`, however I have some issues with building those two images, looks like C++ stdlib panics.

**Comment by aknuds1** [comment]
I can't even build the grafana/build-container Docker image on master branch, due to compilation failure for the osxcross dependency. I wonder what changed :/

**Comment by marefr** [comment]
Added #20413 to PR description

**Comment by aknuds1** [comment]
I've upgraded the grafana/build-container Docker image to Ubuntu 18.04 now, and also Go 1.13.4, pushed it as v1.2.12.

**Comment by marefr** [comment]
@Estrax think you need to update this to get a successful build:
https://github.com/grafana/grafana/blob/2079386a7d15b572bcb7fff272f24002eab88bb5/package.json#L278

**Comment by Estrax** [comment]
@marefr I think I already updated it: https://github.com/grafana/grafana/pull/20405/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2

**Comment by marefr** [comment]
@Estrax I see. Okay so a new build container version needs to be built and pushed to docker hub now when fixed in master and you've merged that in. Let us know if you want me or @aknuds1 to help out with this.

**Comment by Estrax** [comment]
@marefr @aknuds1 built a new container and pushed it as v1.2.13.

**Review by aknuds1 — APPROVED** [review]
LGTM!

**Review by marefr — APPROVED** [review]
LGTM

**Comment by marefr** [comment]
Added an issue to upgrade go in enterprise https://github.com/grafana/grafana-enterprise/issues/298

**Comment by marefr** [comment]
Moved this to 6.6 milestone

**Pull request merged**